src/main/java/io/meeds/qa/ui/pages/page/factory/space/ManageSpacesPage.java

```diff
@@ -657,10 +637,53 @@ public void uploadSpaceBanner(String fileName) {
     builder.moveToElement(spaceBanner).build().perform();
     uploadSpaceBannerButton.waitUntilVisible();
     WebElement elem =
-                    getDriver().findElement(org.openqa.selenium.By.xpath("//*[@class='v-input__prepend-outer']//button/following::input[1]"));
+            getDriver().findElement(org.openqa.selenium.By.xpath("//*[@class='v-input__prepend-outer']//button/following::input[1]"));
     String js = "arguments[0].style.height='auto'; arguments[0].style.visibility='visible';";
     ((JavascriptExecutor) getDriver()).executeScript(js, elem);
     upload(UPLOAD_DIRECTORY_PATH + fileName).fromLocalMachine().to(elem);
   }
 
-}
+  public void removeApplication(String appName) {
+    getApplicationToRemove(appName).clickOnElement();
+    getRemoveOption(appName).clickOnElement();
+    removeConfirmationButton.clickOnElement();
+  }
+
+  public void moveAfterAppButtonIsDisplayed(String appName) {
+    assertWebElementVisible(moveAfterAppButton(appName));
+  }
+
+  public void moveBeforeAppButtonIsDisplayed(String appName) {
+    assertWebElementVisible(moveAfterAppButton(appName));
+  }
+
+  public void removeAppButtonIsDisplayed(String appName) {
+    assertWebElementVisible(getRemoveOption(appName));
+  }
+
+  private BaseElementFacade getApplicationToRemove(String appName) {
+    return findByXPathOrCSS(String.format("//*[contains(@class,'SpaceApplicationCardTitle' )and contains(text(),'%s')]//following::*[@class='v-icon notranslate mdi mdi-dots-vertical theme--light']",
+            appName));
+  }
+
+  private BaseElementFacade getRemoveOption(String appName) {
+    return findByXPathOrCSS(String.format("//*[contains(text(),'%s')]//ancestor::*[contains(@class, 'SpaceApplicationCard')]//button//following::*[contains(@class,'v-list--dense')]//*[contains(text(),'Remove')]",
+            appName));
+  }
+
+  private BaseElementFacade moveBeforeAppButton(String appName) {
+    return findByXPathOrCSS(String.format("//*[contains(text(),'%s')]//ancestor::*[contains(@class, 'SpaceApplicationCard')]//button//following::*[contains(@class,'v-list--dense')]//*[contains(text(),'Move before')]",
+            appName));
+  }
+
+  private BaseElementFacade moveAfterAppButton(String appName) {
+    return findByXPathOrCSS(String.format("//*[contains(text(),'%s')]//ancestor::*[contains(@class, 'SpaceApplicationCard')]//button//following::*[contains(@class,'v-list--dense')]//*[contains(text(),'Move after')]",
+            appName));
+  }
+
+  private BaseElementFacade getApplicationThreeDotsMenu(String appName) {
+    return findByXPathOrCSS(String.format("//*[contains(text(),'%s')]//ancestor::*[contains(@class, 'SpaceApplicationCard')]//button",
```

Comment:
More specific XPath in case new buttons are added in card

```suggestion
    return findByXPathOrCSS(String.format("//*[contains(text(),'%s')]//ancestor::*[contains(@class, 'SpaceApplicationCard')]//*[contains(@class, 'mdi-dots-vertical')]//ancestor::button",
```

Reply:
Done